drivers/sensor/hx711/Kconfig

```diff
@@ -0,0 +1,93 @@
+# HX711 Weight Scale interface Sensor configuration options
+# Copyright (c) 2020, George Gkinis
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig HX711
+	bool "HX711 Weight Scale Sensor"
+	depends on GPIO
+	help
+	  Enable driver for the HX711 Weight Scale Sensor.
+
+if HX711
+
+config HX711_OFFSET
+	int "Initial Offset"
+	default 0
+	help
+	  Sensor offset.
+
+config HX711_SLOPE_INTEGER
+	int "Initial Slope. Integer part."
+	default 1
+	help
+	  Sensor slope, integer part.
+
+config HX711_SLOPE_DECIMAL
+	int "Initial Slope. Decimal part."
+	default 0
+	help
+	  Sensor slope, decimal part.
+	  Multiply by 100000. For .51 enter 510000
+
+config HX711_CALIBRATION_WEIGHT_INTEGER
+	int "Calibration weight."
+	default 0
+	help
+	  Known weight for calibration purposes, integer part.
+
+config HX711_CALIBRATION_WEIGHT_DECIMAL
+	int "Calibration weight."
+	default 0
+	help
+	  Known weight for calibration purposes, decimal part.
+	  Multiply by 100000. For .51 enter 510000
+
+config HX711_SAMPLE_FETCH_TIMEOUT_MS
+	int "Timeout for waiting until reading is ready."
+	default 700
+	help
+	  Fill in the time needed for the HX711 module to be ready after Power Up.
+
+choice
+	prompt "HX711 sampling rate"
+	default HX711_RATE_10HZ
+	help
+	  Select sampling rate.
+config HX711_RATE_10HZ
+	bool "10Hz"
+config HX711_RATE_80HZ
+	bool "80Hz"
+endchoice
+
+config HX711_SAMPLING_RATE
+	int "HX711 sampling rate"
+	default 0
+	default 0 if HX711_RATE_10HZ
+	default 1 if HX711_RATE_80HZ
+
+choice
+	prompt "HX711 gain"
+	default HX711_GAIN_128X
+	help
+	  Select HX711 gain.
+config HX711_GAIN_32X
+	bool "32x"
+config HX711_GAIN_64X
+	bool "64x"
+config HX711_GAIN_128X
+	bool "128x"
+endchoice
+
+config HX711_DISABLE_INTERRUPTS_WHILE_POLLING
+	bool "Disable interrupts while polling"
+	help
+	  Disable interrupts while polling for data.
+
+config HX711_GAIN
+	int
+	default 1
+	default 1 if HX711_GAIN_128X
+	default 2 if HX711_GAIN_32X
+	default 3 if HX711_GAIN_64X
```

Comment:
Please move Kconfig parameters to DT

Reply:
This is not a HW configuration and can change in runtime.
Why should it be in DT?

Reply:
FYI https://docs.zephyrproject.org/latest/guides/dts/dt-vs-kconfig.html#devicetree-versus-kconfig

Reply:
As stated in the documentation initial configuration parameters should reside in DT.
Fixed and added getters and settters for all parameters.

Reply:
Done.